sdk/src/main/java/io/radar/sdk/util/RadarSimpleLogBuffer.kt

```diff
@@ -3,59 +3,189 @@ package io.radar.sdk.util
 import io.radar.sdk.Radar
 import io.radar.sdk.model.RadarLog
 import java.util.concurrent.LinkedBlockingDeque
+import android.content.Context
+import io.radar.sdk.RadarSettings
+import org.json.JSONObject
+import java.io.File
+import java.lang.Integer.min
+import java.util.Date
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
 
 /**
- * Basic Log Buffer implementation that is backed by a field.
+ * Log Buffer implementation that persist.
  */
-internal class RadarSimpleLogBuffer : RadarLogBuffer {
+internal class
+RadarSimpleLogBuffer(override val context: Context): RadarLogBuffer{
 
     private companion object {
-        const val MAXIMUM_CAPACITY = 1000
+
+        const val MAXIMUM_MEMORY_CAPACITY = 200
+
+        const val MAXIMUM_CAPACITY = 500
         const val HALF_CAPACITY = MAXIMUM_CAPACITY / 2
+        const val logFileDir = "radar_logs"
+        var counter = 0
+
     }
+    private var featureFlag = false
 
-    /**
-     * Concurrency-safe list of logs with a maximum capacity of 1000
-     */
-    private val list = LinkedBlockingDeque<RadarLog>(MAXIMUM_CAPACITY)
 
-    override fun write(level: Radar.RadarLogLevel, message: String, type: Radar.RadarLogType?) {
-        if (!list.offer(RadarLog(level, message, type))) {
-            purgeOldestLogs()
-            list.put(RadarLog(level, message, type))
+    private val scheduler = Executors.newScheduledThreadPool(1)
+
+    init {
+        scheduler.scheduleAtFixedRate({ persistLogs() }, 30, 30, TimeUnit.SECONDS)
```

Comment:
@KennyHuRadar @lmeier How are other SDK's flushing to disk? If they do schedule it, how frequently are they flushing?

Reply:
Looking at Segment, it seems like every time there's a call of one of their main functions (`identify`, `track`, `screen`), they `enqueueAction`: https://github.com/segmentio/analytics-ios/blob/23be21248a147a551f0cd7bfb24a6b2df13b1d90/Segment/Classes/SEGSegmentIntegration.m#L198

which calls `queuePayload` which calls their persistence function. https://github.com/segmentio/analytics-ios/blob/23be21248a147a551f0cd7bfb24a6b2df13b1d90/Segment/Classes/SEGSegmentIntegration.m#L314

Note that they're flushing every 30s.

So in essence, they do it for each invocation of their SDK but the client app.

Reply:
This timer is designed to be the secondary trigger for the flush. The main trigger is supposed to be the in-memory buffer reaching a threshold for a flush. This threshold value was determined via experiments to determine how many logs can be written from memory and disk reliably on backgrounding.